x-pack/plugins/alerting/server/task_runner/create_execution_handler.ts

```diff
@@ -112,19 +114,35 @@ export function createExecutionHandler<
         }),
       }));
 
-    alertExecutionStore.numberOfScheduledActions += actions.length;
+    alertExecutionStore.incrementNumberOfScheduledActions(actions.length);
 
     const ruleLabel = `${ruleType.id}:${ruleId}: '${ruleName}'`;
 
     const actionsClient = await actionsPlugin.getActionsClientWithRequest(request);
     let ephemeralActionsToSchedule = maxEphemeralActionsPerRule;
 
     for (const action of actions) {
-      if (alertExecutionStore.numberOfTriggeredActions >= ruleType.config!.execution.actions.max) {
-        alertExecutionStore.triggeredActionsStatus = ActionsCompletion.PARTIAL;
+      if (alertExecutionStore.hasReachedTheExecutableActionsLimit(actionsConfigMap)) {
+        alertExecutionStore.setTriggeredActionsStatus(ActionsCompletion.PARTIAL);
+        logger.debug(
```

Comment:
We need to print the rule id - you should be able to find some similar logging, I can't remember what we typically print from the rule - certainly the id, and I think the space, sometimes the name?